File: tictoctoe.py
board = [[' ',' ',' '],
         [' ',' ',' '],
         [' ',' ',' ']]

def print_board():
    for row in board:
        print(row)

def check_win(player):
    # check rows
    for row in board:
        if row == [player, player, player]:
            return True
    # check columns
    for col in range(3):
        if board[0][col] == player and board[1][col] == player and board[2][col] == player:
            return True
    # check diagonals
    if board[0][0] == player and board[1][1] == player and board[2][2] == player:
        return True
    if board[0][2] == player and board[1][1] == player and board[2][0] == player:
        return True
    return False

def tic_tac_toe():
    player = 'X'
    while True:
        print_board()
        move = input("Player %s, enter your move (row col): " % player)
        row, col = move.split()
        row, col = int(row), int(col)
        if board[row][col] != ' ':
            print("This space is already occupied. Try again.")
            continue
        board[row][col] = player
        if check_win(player):
            print_board()
            print("Player %s wins!" % player)
            break
        if player == 'X':
            player = 'O'
        else:
            player = 'X'

tic_tac_toe()

File: tictoctoe.cpp
#include <iostream>
using namespace std;

char board[3][3] = {{' ',' ',' '},{' ',' ',' '},{' ',' ',' '}};

void print_board()
{
    for(int i=0;i<3;i++)
    {
        for(int j=0;j<3;j++)
            cout << board[i][j] << " ";
        cout << endl;
    }
}

bool check_win(char player)
{
    // check rows
    for(int i=0;i<3;i++)
    {
        if(board[i][0]==player && board[i][1]==player && board[i][2]==player)
            return true;
    }
    // check columns
    for(int i=0;i<3;i++)
    {
        if(board[0][i]==player && board[1][i]==player && board[2][i]==player)
            return true;
    }
    // check diagonals
    if(board[0][0]==player && board[1][1]==player && board[2][2]==player)
        return true;
    if(board[0][2]==player && board[1][1]==player && board[2][0]==player)
        return true;
    return false;
}

int main()
{
    char player = 'X';
    while(true)
    {
        print_board();
        int row,col;
        cout << "Player " << player << ", enter your move (row col): ";
        cin >> row >> col;
        if(board[row][col] != ' ')
        {
            cout << "This space is already occupied. Try again." << endl;
            continue;
        }
        board[row][col] = player;
        if(check_win(player))
        {
            print_board();
            cout << "Player " << player << " wins!" << endl;
            break;
        }
        if(player == 'X')
            player = 'O';
        else
            player = 'X';
    }
    return 0;
}
